src/dmd/declaration.d

```diff
@@ -345,10 +345,11 @@ extern (C++) abstract class Declaration : Dsymbol
      *  e1   = `null` or `this` expression when this declaration is a field
      *  sc   = context
      *  flag = !=0 means do not issue error message for invalid modification
+     * fass  = `true` if this is a field, but a subfield of this is actually modified
      * Returns:
      *  Modifiable.yes or Modifiable.initialization
      */
-    extern (D) final Modifiable checkModify(Loc loc, Scope* sc, Expression e1, int flag)
+    extern (D) final Modifiable checkModify(Loc loc, Scope* sc, Expression e1, int flag, bool fass = false)
```

Comment:
Only 1 bit of the `int flag` parameter is actually used.  You could pass this information in the 2nd bit.

Reply:
But that would require to modify all usages of flag to check for the first bit.